src/Enable.Extensions.Queuing.RabbitMQ/Internal/RabbitMQQueueClient.cs

```diff
@@ -144,53 +139,29 @@ public RabbitMQQueueClient(ConnectionFactory connectionFactory, string queueName
             Func<IQueueMessage, CancellationToken, Task> messageHandler,
             MessageHandlerOptions messageHandlerOptions)
         {
+            lock (_channel)
+            {
+                // Reconfigure QoS in order to support specified level of concurrency.
+                // Here we are assuming a "worker queue", where multiple consumers are
+                // competing to draw from a single queue in order to spread workload
+                // across the consumers.
+                _channel.BasicQos(
```

Comment:
It is a shame to have to call invoke `_channel.BasicQos` every time `RegisterMessageHandler` is called. I couldn't see any alternatives given the current API design.